### PR TITLE
Optimization logic of data storage and data loading

### PR DESCRIPTION
**Pull request by BogdanChernikov**

### CODE REVIEW

File: ExpensesTracker/Forms/MainWindow.Designer.cs [hunk 1]
```diff
@@ -198,7 +201,7 @@ public void InitializeComponent()
             this.accountBox.Name = "accountBox";
             this.accountBox.Size = new System.Drawing.Size(157, 21);
             this.accountBox.TabIndex = 10;
-            this.accountBox.SelectedIndexChanged += new System.EventHandler(this.SelectedAccountBox_SelectedIndexChanged);
+            this.accountBox.SelectedIndexChanged += (SelectedAccountBox_SelectedIndexChanged);
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.Designer.cs]: Change format. Do not modify this file manualy.

File: ExpensesTracker/Forms/MainWindow.cs [hunk 2]
```diff
@@ -27,15 +27,32 @@ public MainWindow()
         private void MainWindow_Load(object sender, EventArgs e)
         {
             _accounts = _storage.GetAccounts();
-            _accountOperations = _storage.GetAccountOperations();
+
+            CreateDefaultAccount();
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs]: Check if any account exists before calling this method.

File: ExpensesTracker/Forms/MainWindow.cs [hunk 3]
```diff
@@ -27,15 +27,32 @@ public MainWindow()
...
 
             accountBox.DataSource = null;
             accountBox.DataSource = _accounts;
             accountBox.DisplayMember = "Name";
-            accountBox.SelectedIndex = 0;
 
             operationsTable.AutoGenerateColumns = false;
+
+            accountBox.SelectedIndex = 0;
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs:40]: Subscribe method on SelectedIndexChanged event after you set SelectedIndex as "0". Call ShowTable and ShowBalance method explicitly.

File: ExpensesTracker/Forms/MainWindow.cs [hunk 4]
```diff
@@ -27,15 +27,32 @@ public MainWindow()
...
+
             categoryFilterBox.SelectedItem = "All amount";
+            ShowBalance();
+        }
+
+        private void CreateDefaultAccount()
+        {
+            if (_accounts.Count == 0)
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs]: Remove this condition from here.

File: ExpensesTracker/Forms/MainWindow.cs [hunk 5]
```diff
@@ -91,10 +106,10 @@ private void FilterOperations()
             _filteredOperations = _filteredOperations.Where(x => x.Date <= endDateDisplay.Value).ToList();
         }
 
-        private void RefreshData()
+        public void RefreshData()
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs]: Make it private

File: ExpensesTracker/Forms/MainWindow.cs [hunk 6]
```diff
@@ -91,10 +106,10 @@ private void FilterOperations()
...
         {
+            TargetAccount.AccountOperations = AccountOperations;
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs]: Remove this line :)

File: ExpensesTracker/Forms/MainWindow.cs (hunk 3 above)
Comment (VolodymyrMaryniak) [reply ExpensesTracker/Forms/MainWindow.cs:41]: You don't need to unsubscribe, do an operation, and then subscribe again. 
Just remove a subscription from MainWindows.Designer and subscribe after the operation.

File: ExpensesTracker/Forms/MainWindow.Designer.cs [hunk 7]
```diff
@@ -198,8 +201,7 @@ public void InitializeComponent()
             this.accountBox.Name = "accountBox";
             this.accountBox.Size = new System.Drawing.Size(157, 21);
             this.accountBox.TabIndex = 10;
-            this.accountBox.SelectedIndexChanged += new System.EventHandler(this.SelectedAccountBox_SelectedIndexChanged);
-            // 
+            this.accountBox.SelectedIndexChanged += new System.EventHandler(this.SelectedAccountBox_SelectedIndexChanged);            //
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.Designer.cs]: Check this. Move this "// " to the next line.

File: ExpensesTracker/Forms/MainWindow.cs [hunk 8]
```diff
@@ -27,15 +27,34 @@ public MainWindow()
...
             accountBox.DisplayMember = "Name";
-            accountBox.SelectedIndex = 0;
 
             operationsTable.AutoGenerateColumns = false;
+
+            accountBox.SelectedIndexChanged -= (SelectedAccountBox_SelectedIndexChanged);
+            accountBox.SelectedIndex = 0;
+            accountBox.SelectedIndexChanged += (SelectedAccountBox_SelectedIndexChanged);
+
             categoryFilterBox.SelectedItem = "All amount";
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs:43]: Remove a subscription from MainWindows.Designer and subscribe after the operation.

File: ExpensesTracker/Forms/MainWindow.cs [hunk 9]
```diff
@@ -139,27 +154,34 @@ private void SavePdfButton_Click(object sender, EventArgs e)
...
+            accountBox.SelectedIndexChanged -= (SelectedAccountBox_SelectedIndexChanged);
             accountBox.DataSource = null;
             accountBox.DataSource = _accounts;
             accountBox.DisplayMember = "Name";
-            accountBox.SelectedItem = TargetAccount;
+            accountBox.SelectedIndexChanged += (SelectedAccountBox_SelectedIndexChanged);
+            accountBox.SelectedItem = _accounts.First();
         }
 
         public void ShowBalance()
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs]: Make the next methods private:
SelectedAccountBox_SelectedIndexChanged
ActualizeTableRecords
ShowBalance

Comment (VolodymyrMaryniak) [reply ExpensesTracker/Forms/MainWindow.cs]: Do methods private until otherwise is required.

File: ExpensesTracker/Forms/MainWindow.cs [hunk 10]
```diff
@@ -139,27 +154,34 @@ private void SavePdfButton_Click(object sender, EventArgs e)
 
         private void RefreshAccountsData()
         {
+            accountBox.SelectedIndexChanged -= (SelectedAccountBox_SelectedIndexChanged);
```
Comment (VolodymyrMaryniak) [inline ExpensesTracker/Forms/MainWindow.cs:156]: Is it required here? 
Just asking. If it's required then keep it.

Comment (BogdanChernikov) [reply ExpensesTracker/Forms/MainWindow.cs:156]: yes, its required

**Pull request merged**